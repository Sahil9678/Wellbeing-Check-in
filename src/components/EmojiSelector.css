.emoji-selector {
    text-align: center;
    padding: 20px;
}

.emoji-container {
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
}

.emoji-item {
    cursor: pointer;
    margin: 10px;
    padding: 10px;
    border: 2px solid transparent;
    border-radius: 10px;
    transition: border-color 0.3s ease;
}

.emoji-item:hover {
    border-color: #ccc;
}

.emoji-item.selected {
    border-color: #007bff;
}

.emoji-item span {
    display: block;
    font-size: 2rem;
}

.emoji-item p {
    margin: 5px 0 0;
    font-size: 1rem;
}

@media (max-width: 768px) {
    .emoji-item span {
        font-size: 1.5rem;
    }

    .emoji-item p {
        font-size: 0.8rem;
    }
}

@media (max-width: 480px) {
    .emoji-item span {
        font-size: 1.2rem;
    }

    .emoji-item p {
        font-size: 0.7rem;
    }
}