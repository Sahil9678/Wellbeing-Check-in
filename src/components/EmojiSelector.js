import React, { useState } from 'react';
import './EmojiSelector.css';

const emojis = [
    { emoji: '😞', label: 'Terrible' },
    { emoji: '😟', label: 'Bad' },
    { emoji: '😐', label: 'Alright' },
    { emoji: '😊', label: 'Pretty Good' },
    { emoji: '😁', label: 'Fantastic' }
];

const EmojiSelector = () => {
    const [selectedEmoji, setSelectedEmoji] = useState(null);

    const handleClick = (label) => {
        setSelectedEmoji(label);
    };

    return (
        <div className="emoji-selector">
            <h2>Wellbeing Check-in</h2>
            <p>Hello! How are you feeling today?</p>
            <div className="emoji-container">
                {emojis.map((item) => (
                    <div
                        key={item.label}
                        className={`emoji-item ${selectedEmoji === item.label ? 'selected' : ''}`}
                        onClick={() => handleClick(item.label)}
                    >
                        <span role="img" aria-label={item.label}>
                            {item.emoji}
                        </span>
                        <p>{item.label}</p>
                    </div>
                ))}
            </div>
        </div>
    );
};

export default EmojiSelector;