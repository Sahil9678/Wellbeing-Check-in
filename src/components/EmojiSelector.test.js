import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import EmojiSelector from './EmojiSelector';

describe('EmojiSelector', () => {
    //First test case
    it('should render the component with a heading and prompt', () => {
        render(<EmojiSelector />);
        
        // 1.Verify  heading
        const headingElement = screen.getByText(/Wellbeing Check-in/i);
        expect(headingElement).toBeInTheDocument();

        // 2. Verify prompt text
        const promptElement = screen.getByText(/Hello! How are you feeling today?/i);
        expect(promptElement).toBeInTheDocument();
    });

    //Second test case
    it('should render all emojis with their labels', () => {
        render(<EmojiSelector />);
        
        const emojis = [
            { emoji: '😞', label: 'Terrible' },
            { emoji: '😟', label: 'Bad' },
            { emoji: '😐', label: 'Alright' },
            { emoji: '😊', label: 'Pretty Good' },
            { emoji: '😁', label: 'Fantastic' }
        ];

        //3. Verify each emoji and its label is rendered
        emojis.forEach(({ emoji, label }) => {
            const emojiElement = screen.getByRole('img', { name: label });
            const labelElement = screen.getByText(label);

            expect(emojiElement).toBeInTheDocument();
            expect(emojiElement).toHaveTextContent(emoji);
            expect(labelElement).toBeInTheDocument();
        });
    });

    //Third test case
    it('should update the selected emoji on click', () => {
        render(<EmojiSelector />);
        
        const firstEmoji = screen.getByText('Terrible');
        
        // Click the first emoji
        fireEvent.click(firstEmoji);
        
        // Check if the first emoji is selected
        expect(firstEmoji.closest('.emoji-item')).toHaveClass('selected');
        
        const secondEmoji = screen.getByText('Bad');
        
        // Click the second emoji
        fireEvent.click(secondEmoji);
        
        // Check if the second emoji is selected
        expect(secondEmoji.closest('.emoji-item')).toHaveClass('selected');
        
        // Ensure the first emoji is no longer selected
        expect(firstEmoji.closest('.emoji-item')).not.toHaveClass('selected');
    });

    //Fourth test case
    it('should match the snapshot', () => {
        const { asFragment } = render(<EmojiSelector />);
        expect(asFragment()).toMatchSnapshot();
    });
});
