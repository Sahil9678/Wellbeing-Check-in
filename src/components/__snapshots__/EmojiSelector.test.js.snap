// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmojiSelector should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="emoji-selector"
  >
    <h2>
      Wellbeing Check-in
    </h2>
    <p>
      Hello! How are you feeling today?
    </p>
    <div
      class="emoji-container"
    >
      <div
        class="emoji-item "
      >
        <span
          aria-label="Terrible"
          role="img"
        >
          😞
        </span>
        <p>
          Terrible
        </p>
      </div>
      <div
        class="emoji-item "
      >
        <span
          aria-label="Bad"
          role="img"
        >
          😟
        </span>
        <p>
          Bad
        </p>
      </div>
      <div
        class="emoji-item "
      >
        <span
          aria-label="Alright"
          role="img"
        >
          😐
        </span>
        <p>
          Alright
        </p>
      </div>
      <div
        class="emoji-item "
      >
        <span
          aria-label="Pretty Good"
          role="img"
        >
          😊
        </span>
        <p>
          Pretty Good
        </p>
      </div>
      <div
        class="emoji-item "
      >
        <span
          aria-label="Fantastic"
          role="img"
        >
          😁
        </span>
        <p>
          Fantastic
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
